### PR TITLE
[22.01] Update interactive tool docs for Gunicorn / dropping uWSGI

### PR DESCRIPTION
**Pull request by mvdbeek**
xref #12257 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by mvdbeek** [comment]
This goes together with https://github.com/galaxyproject/gravity/pull/32

File: config/galaxy.yml.interactivetools [hunk 1]
```diff
@@ -1,54 +1,19 @@
...
   interactivetools_map: database/interactivetools_map.sqlite
   # If you develop InteractiveTools locally and do not have a full FQDN you can
-  # use an arbritrary one, e.g. 'my-hostname' here, if you set this hostname in your
-  # job_conf.xml as well (see the corresponding comment). If running mac OS X, do not match the
-  # "http://host.docker.internal:8080" used at galaxy_infrastructure_url in the galaxy.yml file, 
-  # and use an arbitrary name here instead.
-  # Please make sure that in the local development case you use https://localhost:8080 to access
-  # your Galaxy. http://my-hostname:8080 will not work.
-  # galaxy_infrastructure_url: http://my-hostname:8080
+  # use an arbritrary one, e.g. 'host.docker.internal' here, if you set this hostname in your
```
Comment (nsoranzo) [inline config/galaxy.yml.interactivetools]:
```suggestion
  # use an arbitrary one, e.g. 'host.docker.internal' here, if you set this hostname in your
```

File: config/job_conf.xml.interactivetools [hunk 2]
```diff
@@ -21,21 +20,16 @@
                If you develop interactive tools on your 'localhost' and don't have a proper domain name
                you need to tell all Docker containers a hostname where Galaxy is running.
                This can be done via the add-host parameter during the `docker run` command.
-               'my-hostname' here is an arbritrary hostname that matches the IP address of your
-               Galaxy host. Make sure this hostname ('my-hostname') is also set in your galaxy.yml file, e.g.
-               `galaxy_infrastructure_url: http://my-hostname:8080`.
-              Note: XML does not allow double hyphens in comments, hence the line below has them escaped as &#45;
+               'localhost' here is an arbritrary hostname that matches the IP address of your
```
Comment (nsoranzo) [inline config/job_conf.xml.interactivetools]:
```suggestion
               'localhost' here is an arbitrary hostname that matches the IP address of your
```

File: doc/source/admin/galaxy_options.rst [hunk 3]
```diff
@@ -1707,13 +1707,25 @@
 :Type: bool
 
 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``interactivetools_upstream_proxy``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:Description:
+    Set this to false to redirect users of Interacitve tools directly
```
Comment (nsoranzo) [inline doc/source/admin/galaxy_options.rst]:
```suggestion
    Set this to false to redirect users of Interactive tools directly
```

File: lib/galaxy/config/sample/galaxy.yml.sample [hunk 4]
```diff
@@ -921,8 +921,13 @@ galaxy:
   # Enable InteractiveTools.
   #interactivetools_enable: false
 
-  # Proxy host - assumed to just be hosted on the same hostname and port
-  # as Galaxy by default.
+  # Set this to false to redirect users of Interacitve tools directly to
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/galaxy.yml.sample]:
```suggestion
  # Set this to false to redirect users of Interactive tools directly to
```

File: lib/galaxy/config/schemas/config_schema.yml [hunk 5]
```diff
@@ -1231,11 +1231,19 @@ mapping:
         desc: |
           Enable InteractiveTools.
 
+      interactivetools_upstream_proxy:
+        type: bool
+        default: true
+        required: false
+        desc:
+          Set this to false to redirect users of Interacitve tools directly to the Interactive tools proxy.
```
Comment (nsoranzo) [inline lib/galaxy/config/schemas/config_schema.yml]:
```suggestion
          Set this to false to redirect users of Interactive tools directly to the Interactive tools proxy.
```

File: doc/source/admin/special_topics/interactivetools.rst [hunk 6]
```diff
@@ -96,22 +118,35 @@ An example ``job_conf.xml`` file as seen in ``config/job_conf.xml.interactivetoo
...
                   <param id="docker_enabled">true</param>
+                  <!-- If you have not set 'outputs_to_working_directory: true' in galaxy.yml you can remove the docker_volumes setting. -->
                   <param id="docker_volumes">$galaxy_root:ro,$tool_directory:ro,$job_directory:rw,$working_directory:rw,$default_file_path:ro</param>
                   <param id="docker_sudo">false</param>
                   <param id="docker_net">bridge</param>
                   <param id="docker_auto_rm">true</param>
                   <param id="require_container">true</param>
+                  <param id="container_monitor">true</param>
+                  <param id="docker_set_user"></param>
+                  <!-- InteractiveTools do need real hostnames or URLs to work - simply specifying IPs will not work.
```
Comment (nsoranzo) [inline doc/source/admin/special_topics/interactivetools.rst]:
```suggestion
                  <!-- InteractiveTools need real hostnames or URLs to work - simply specifying IPs will not work.
```

File: doc/source/admin/special_topics/interactivetools.rst [hunk 7]
```diff
@@ -96,22 +118,35 @@ An example ``job_conf.xml`` file as seen in ``config/job_conf.xml.interactivetoo
...
+                       If you develop interactive tools on your 'localhost' and don't have a proper domain name
+                       you need to tell all Docker containers a hostname where Galaxy is running.
+                       This can be done via the add-host parameter during the `docker run` command.
+                       'localhost' here is an arbritrary hostname that matches the IP address of your
```
Comment (nsoranzo) [inline doc/source/admin/special_topics/interactivetools.rst]:
```suggestion
                       'localhost' here is an arbitrary hostname that matches the IP address of your
```

File: config/job_conf.xml.interactivetools [hunk 8]
```diff
@@ -21,21 +20,16 @@
...
-               Galaxy host. Make sure this hostname ('my-hostname') is also set in your galaxy.yml file, e.g.
-               `galaxy_infrastructure_url: http://my-hostname:8080`.
-              Note: XML does not allow double hyphens in comments, hence the line below has them escaped as &#45;
+               'localhost' here is an arbitrary hostname that matches the IP address of your
+               Galaxy host. Make sure this hostname ('localhost') is also set in your galaxy.yml file, e.g.
+               `galaxy_infrastructure_url: http://localhost:8080`.
           -->
-          <!-- <param id="docker_run_extra_arguments">&#45;&#45;add-host my-hostname:10.1.2.3</param> -->
-
+          <param id="docker_run_extra_arguments">--add-host localhost:host-gateway</param>
```
Comment (bgruening) [inline config/job_conf.xml.interactivetools:27]: "host.docker.internal" maybe to match the example from the galaxy.yml?

Comment (mvdbeek) [reply config/job_conf.xml.interactivetools:27]: Which example ? The hostname needs to match `galaxy_infrastructure_url`, which is set to `http://localhost:8080` in galaxy.yml.interactivetools. I feel like `host.docker.internal` gives you the idea that this is some magic address that can't be changed.

Comment (bgruening) [reply config/job_conf.xml.interactivetools:27]: https://github.com/galaxyproject/galaxy/pull/13470/files#diff-dc06bbcb59cc0db161563c300c1084e592aead949dbdc6f88b4687040344662cR12  this here ..

Comment (mvdbeek) [reply config/job_conf.xml.interactivetools:27]: Alright, I missed that. Better in https://github.com/galaxyproject/galaxy/pull/13470/commits/ed8ce47c42f6fe216409907dfec31d0b8811be7b ?

**Comment by jmchilton** [comment]
Thank for so much for this dedication to the docs, this is incredible work.

**Pull request merged**